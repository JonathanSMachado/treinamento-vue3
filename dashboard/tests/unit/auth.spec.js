import mockAxios from 'axios'
import AuthService from '../../src/services/auth'

jest.mock('axios')

describe('AurhService', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should return a token when user login', async () => {
    const token = '123.123.123'
    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({ data: { token } })
    })

    const response = await AuthService(mockAxios).login({ email: 'jonathan@gmail.com', password: '1234' })
    expect(response.data).toHaveProperty('token')
    expect(response).toMatchSnapshot()
  })

  it('should return on user when user register', async () => {
    const user = {
      name: 'Jonathan',
      password: '1234',
      email: 'jonathan@gmail.com'
    }

    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({ data: user })
    })

    const response = await AuthService(mockAxios).register(user)
    expect(response.data).toHaveProperty('name')
    expect(response.data).toHaveProperty('password')
    expect(response.data).toHaveProperty('email')
    expect(response.data).toMatchSnapshot()
  })
})
