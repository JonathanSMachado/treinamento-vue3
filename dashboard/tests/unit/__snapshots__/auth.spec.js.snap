// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AurhService should return a token when user login 1`] = `
Object {
  "data": Object {
    "token": "123.123.123",
  },
  "errors": null,
}
`;

exports[`AurhService should return on user when user register 1`] = `
Object {
  "email": "jonathan@gmail.com",
  "name": "Jonathan",
  "password": "1234",
}
`;
